Cubical/Categories/Constructions/Slice.agda

```diff
@@ -0,0 +1,383 @@
+                           (cong extractIso' (JRefl pToIFam' pToIBase') ∙ sym (JRefl pToIFam pToIBase))
+                           p
+               where
+                 idx = C .id x
+                 pToIFam = (λ z _ → CatIso {C = C} x z)
+                 pToIBase = catiso (C .id x) idx (C .⋆IdL idx) (C .⋆IdL idx)
+
+                 idxf = SliceCat .id xf
+                 pToIFam' = (λ z _ → CatIso {C = SliceCat} xf z)
+                 pToIBase' = catiso (SliceCat .id xf) idxf (SliceCat .⋆IdL idxf) (SliceCat .⋆IdL idxf)
+
+            -- why does this not follow definitionally?
+            -- from extractCom, we get that performing the roundtrip on pOb gives us back p'Ob
+            ppp : p'Ob ≡ (pToIIso .inv) (pToIIso .fun pOb)
+            ppp = cong (pToIIso .inv) extractCom
+
+            -- apply univalence of C
+            -- this gives us the first component that we want
+            p'Ob≡pOb : p'Ob ≡ pOb
+            p'Ob≡pOb = ppp ∙ pToIIso .leftInv pOb
+
+            -- isSetHom gives us the second component, path between morphisms
+            p'Mor≡pMor : PathP (λ j → PathP (λ i → C [ (p'Ob≡pOb j) i , c ]) f g) p'Mor pMor
+            p'Mor≡pMor = isSetHomP2l _ _ p'Mor pMor p'Ob≡pOb
+
+            -- we can use the above paths to show that p' ≡ p
+            p'≡p : p' ≡ p
+            p'≡p i = comp (λ i' → SOPath≡PathΣ {xf = xf} {yg} (~ i'))
+                            (λ j → λ { (i = i0) → left (~ j) ; (i = i1) → right (~ j) })
+                            (p'Σ≡pΣ i)
+              where
+                -- we break up p' and p into their constituent paths
+                -- first via transport and then via our component definitions from before
+                -- we show that p'ΣT ≡ p'Σ (and same for p) via univalence
+                -- and p'Σ≡pΣ follows from our work from above
+                p'ΣT : Σ[ p ∈ x ≡ y ] PathP (λ i → C [ p i , c ]) f g
+                p'ΣT = transport SOPath≡PathΣ p'
+                p'Σ : Σ[ p ∈ x ≡ y ] PathP (λ i → C [ p i , c ]) f g
+                p'Σ = (p'Ob , p'Mor)
+
+                pΣT : Σ[ p ∈ x ≡ y ] PathP (λ i → C [ p i , c ]) f g
+                pΣT = transport SOPath≡PathΣ p
+                pΣ : Σ[ p ∈ x ≡ y ] PathP (λ i → C [ p i , c ]) f g
+                pΣ = (pOb , pMor)-- transport SOPathP≡PathPSO p
+
+                -- using the computation rule to ua
+                p'ΣT≡p'Σ : p'ΣT ≡ p'Σ
+                p'ΣT≡p'Σ = uaβ SOPath≃PathΣ p'
+
+                pΣT≡pΣ : pΣT ≡ pΣ
+                pΣT≡pΣ = uaβ SOPath≃PathΣ p
+
+                p'Σ≡pΣ : p'Σ ≡ pΣ
+                p'Σ≡pΣ = ΣPathP (p'Ob≡pOb , p'Mor≡pMor)
+
+                -- two sides of the square we're connecting
+                left : PathP (λ i → SOPath≡PathΣ {xf = xf} {yg} i) p' p'Σ
+                left = transport-filler SOPath≡PathΣ p' ▷ p'ΣT≡p'Σ
+
+                right : PathP (λ i → SOPath≡PathΣ {xf = xf} {yg} i) p pΣ
+                right = transport-filler SOPath≡PathΣ p ▷ pΣT≡pΣ
+
```

Comment:
It's a bit surprising that this proof is so long!

Reply:
Yeah, it's a bit longer than I suspected... Maybe it's because so many things are given explicit names?